### PR TITLE
Add client listing and management

### PR DESCRIPTION
**Pull request by rklf**

### CODE REVIEW

File: src/app/clients/client-columns.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    accessorKey: 'role',
+    header: ({ column }) => {
+      return (
+        <Button variant="ghost" onClick={() => column.toggleSorting(column.getIsSorted() === 'asc')}>
+          Rôle
+          <ArrowUpDown className="ml-2 h-4 w-4" />
+        </Button>
+      );
+    },
+  },
```
Comment (floperrier) [inline src/app/clients/client-columns.tsx]:
```suggestion
  {
    accessorKey: 'role',
    header: ({ column }) => (
      <div className="flex">
        <Select
          onValueChange={value => {
            if (value === 'all') value = '';
            column.setFilterValue(value || undefined);
          }}
        >
          <SelectTrigger className="w-[180px]">
            <SelectValue placeholder="Tous les rôles" />
          </SelectTrigger>
          <SelectContent>
            <SelectItem value="all">Tous les rôles</SelectItem>
            <SelectItem value="client">Client</SelectItem>
            <SelectItem value="courier">Livreur</SelectItem>
            <SelectItem value="admin">Administrateur</SelectItem>
            <SelectItem value="support">Support</SelectItem>
          </SelectContent>
        </Select>
        <Button variant="ghost" onClick={() => column.toggleSorting(column.getIsSorted() === 'asc')}>
          <ArrowUpDown className="h-4 w-4" />
        </Button>
      </div>
    ),
    enableColumnFilter: true,
  },
```

File: src/app/clients/actions.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button variant="ghost" className="h-8 w-8 p-0">
+          <span className="sr-only">Ouvrir le menu</span>
+          <MoreHorizontal className="h-4 w-4" />
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent align="end">
+        <DropdownMenuLabel>Actions</DropdownMenuLabel>
+        <DropdownMenuItem onClick={() => router.push(`/clients/${client.id}`)}>Voir le client</DropdownMenuItem>
```
Comment (floperrier) [inline src/app/clients/actions.tsx]: This is a CRUD for all users, not only clients, you should maybe change references of "client" to "user"

**Review by floperrier — CHANGES_REQUESTED** [review]

File: src/app/clients/actions.tsx (hunk 2 above)
Comment (rklf) [reply src/app/clients/actions.tsx]: A courier is still a customer of our app though, no?

File: src/app/clients/client-columns.tsx (hunk 1 above)
Comment (rklf) [reply src/app/clients/client-columns.tsx]: Nice, updated.

**Review by floperrier — APPROVED** [review]

**Pull request merged**